packages/trading-client/src/handlers/coordinator.ts

```diff
@@ -0,0 +1,245 @@
+import { CoordinatorContract, CoordinatorRegistryContract, ExchangeContract } from '@0x/abi-gen-wrappers';
+import { assert } from '@0x/assert';
+import { ContractAddresses, getContractAddressesForNetworkOrThrow, NetworkId } from '@0x/contract-addresses';
+import { Coordinator, CoordinatorRegistry, Exchange } from '@0x/contract-artifacts';
+import { TransactionEncoder } from '@0x/contract-wrappers';
+import { schemas } from '@0x/json-schemas';
+import { generatePseudoRandomSalt } from '@0x/order-utils';
+import { SignedOrder } from '@0x/types';
+import { BigNumber, fetchAsync } from '@0x/utils';
+import { Web3Wrapper } from '@0x/web3-wrapper';
+import { SupportedProvider } from 'ethereum-types';
+
+import { Request, RequestHandler } from '../index';
+
+// copied from coordinator server
+enum ExchangeMethods {
+    FillOrder = 'fillOrder',
+    FillOrKillOrder = 'fillOrKillOrder',
+    FillOrderNoThrow = 'fillOrderNoThrow',
+    CancelOrder = 'cancelOrder',
+}
+
+// copied from coordinator server
+enum BatchExchangeMethods {
+    BatchFillOrders = 'batchFillOrders',
+    BatchFillOrKillOrders = 'batchFillOrKillOrders',
+    BatchFillOrdersNoThrow = 'batchFillOrdersNoThrow',
+    MarketSellOrders = 'marketSellOrders',
+    MarketSellOrdersNoThrow = 'marketSellOrdersNoThrow',
+    MarketBuyOrders = 'marketBuyOrders',
+    MarketBuyOrdersNoThrow = 'marketBuyOrdersNoThrow',
+    BatchCancelOrders = 'batchCancelOrders',
+}
+
+const networkInfoByCoordinatorAddress: {
+    [coordinatorAddress: string]: {
+        network: NetworkId;
+        registryAddress: string;
+        exchangeAddress: string;
+    };
+} = {};
+
+Object.keys(NetworkId)
+    .filter(k => typeof NetworkId[k as any] === 'number')
+    .map(name => (NetworkId[name as any] as unknown) as NetworkId)
+    .forEach(name => {
+        const addresses: ContractAddresses = getContractAddressesForNetworkOrThrow(name);
+        networkInfoByCoordinatorAddress[addresses.coordinator] = {
+            network: name,
+            registryAddress: addresses.coordinatorRegistry,
+            exchangeAddress: addresses.exchange,
+        };
+    });
```

Comment:
Is there a way to access contract addresses besides `getContractAddressesForNetworkOrThrow`? That was the only way I found, but for this module, retrieval is always keyed by coordinator contract address. Open to suggestions if there's a better way of doing it!

The `any` hack is because  a string enum returns both keys and values when `Object.keys` is called on it. I can play around with it a bit more to see if there's a different way, and add a comment if I'm keeping it?